src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -509,6 +496,40 @@ public Long getTimeTaken(){
         return timeTaken;       
     }
     
+    /**
+     * Generates the ID for the log. If the googleId is unknown or unregistered, 
+     * the email and course of the {@code student} will be used to construct the id.
+     * @param googleId the google ID
+     * @param student StudentAttributes object
+     * @return log ID
+     */
+    public String generateLogId(String googleId, StudentAttributes student) {
+        if (student != null) {
+            return generateLogId(googleId, student.email, student.course);
+        } else {
+            return generateLogId(googleId, null, null);
+        }
+    }
+    
+    /**
+     * Generates the ID for the log. If the googleId is unknown or unregistered, 
+     * the {@code email} and {@code course} will be used to construct the id.
+     * @param googleId the google ID
+     * @param email the email
+     * @param course the course
+     * @return log ID
+     */
+    public String generateLogId(String googleId, String email, String course) {
```

Comment:
Pass in the time, it is not safe to assume that future maintainers will use the method after having set time before calling this method